CMakeLists.txt

```diff
@@ -632,21 +632,20 @@ set_package_properties(OpenSSL
 		PURPOSE "Provides HTTPS support to libtorrent"
 )
 
-if(OPENSSL_FOUND)
-
+if(TARGET OpenSSL::SSL)
 	# TODO: needed until https://gitlab.kitware.com/cmake/cmake/issues/19263 is fixed
 	if(WIN32 AND OPENSSL_USE_STATIC_LIBS)
 		target_link_libraries(OpenSSL::Crypto INTERFACE crypt32)
 	endif()
 
 	target_link_libraries(torrent-rasterbar PUBLIC OpenSSL::SSL)
 	target_compile_definitions(torrent-rasterbar PUBLIC TORRENT_USE_OPENSSL)
-	target_sources(torrent-rasterbar PRIVATE src/pe_crypto)
 endif()
 
-if(OPENSSL_FOUND)
+if(TARGET OpenSSL::Crypto)
```

Comment:
what does this mean exactly?
is it checking if a target *exists*?

Reply:
assuming that's what it does, whether to link libtorrent against openssl, and use openssl is a user option, right? Isn't it possible that the openssl target could *exist* even though libtorrent is not supposed to link against it?

Reply:
for example, suppose a project using libtorrent pulls it in as a git submodule, and adds a `add)subdirectory()` in their CMakeLists.txt, then adds a dependency to the libtorrent target.

In such scenario, the top level project *could* also pull in an openssl cmake target, without necessarily wanting to build libtorrent against it. Perhaps unlikely, but it seems like those are independent.

Reply:
oh, perhaps I'm wrong. it looks like the CMakeList currently decides whether to link against openssl based on whether it can find it or not...

Reply:
@arvidn 

The current design of libtorrent's CML is such that some features (such as OpenSSL) are enabled by default based on whether or not certain packages are found. In other words, the set of default features is not fixed for all users, it will vary between environments.

This PR does not change anything pertaining to that overall design choice.

Related: Recently, there were some somewhat heated discussions about this in qBittorrent. In the end, the majority decided to change towards a fixed set of default features (that can all be overriden by the user of course), not dependent on found packages in each system.